### ISSUE TITLE
Single vs. Double Quotes

### ISSUE THREAD

**Issue by MPVaccaro** [susanBuck/e2-fall19#31]
In the class examples, ```single quotes``` are used for ```keys``` and ```values```.  I believe you said we could use either ```single quotes``` or ```double quotes``` so long as we were consistent.

I would prefer to use double quotes, as that is how I write ```html code```, and I feel I would be more prone to errors switching back and forth.

Is there a reason that using single quotes is superior to double quotes, or is it truly personal preference?

Since I am a beginner, I would like to develop habits that are **best practice** while also being personally the most **"fool proof."**

**Comment by susanBuck** [comment]
>> I believe you said we could use either single quotes or double quotes so long as we were consistent.

^^ This is correct.

I personally tend towards single quotes because they're less visual clutter and easier to type (don't have to hold a shift key). These are very small points, but as much as they're used throughout code - it can add up!

In general, both will work - with some caveats/things to know:

**Variable expansion**
Double quotes have a nice advantage where variables within are expanded within a string. Example:
```
$message = "Hello $firstName, how are you"?;
```

This above would not work with single quote, so you'd want to join in the variable using the concatenation operator (`.`)

```php
$message = 'Hello ' . $firstName . ', how are you'?;
```

**Speed**
Some argue single quotes are processed faster (because it doesn't have to check to see if there are variables present). Others say the performance difference is negligible, and I tend to agree with that.

**Escape characters**
Double quotes will also work with escape characters - which are character combinations that start with a backslash and have special meaning. E.g. `\n` is the escape characters for a new line:

So the following would work:
```php
$message = "Hello \n world";
```

But this would not:
```php
$message = Hello \n world';
```

**Summary**
Going back to the point of "either single quotes or double quotes so long as we were consistent." - I think it's okay to vary between single and double quotes when it's done purposefully.

For example, a student might choose to default to single quotes for most instances, but use double quotes when joining a variable into a string to take advantage of the variable expansion.

**Comment by RGPhD** [comment]
Dear @MPVaccaro,

Great question Mary V; I was just thinking about this yesterday.  I also recalled Profesora @susanBuck saying "as long as we are consistent".  I tidied my `index.php` file code yesterday to all double quotes like I format my HTML files.  I also tidied my `index-controller.php` file to have all single quotes like in the videos.

I hope that is okay and best practices Profesora @susanBuck.  I was trying to be consistent per file :)

Kind regards,
Robelyn
